### DISCUSSION TITLE
How to save html content with all attributes

### DISCUSSION THREAD

**Discussion by dcr31000** [GrapesJS/grapesjs#3812] [Q&A]
Hello,

I'm using the remote storageManager option to load and save templates. But this don't save the unique id (set by default or by me) on dom components because IDs are stored in "attributes".

Is it possible to retrieve the html with all attributes to parse dom with another tool ?

thanks

**Comment by dcr31000** [comment]
Is it possible to call the method which create the dom content (from gjs-components, gjs-styles, gjs-html.... to dom object) ? Thanks

**Reply by momu-2016** [reply]
save the component and style,but not html and css

**Comment by iabhiyaan** [comment]
to retrive html
component.toHTML(); might work for you...
Reference:
https://grapesjs.com/docs/api/component.html#component
